### PR TITLE
[Merged by Bors] - chore: backports for leanprover/lean4#4814 (part 22)

### PR DESCRIPTION
**Pull request by kim-em**

### CODE REVIEW

File: Mathlib/Analysis/BoxIntegral/Partition/Filter.lean [hunk 1]
```diff
@@ -173,8 +173,7 @@ noncomputable section
 
 namespace BoxIntegral
 
-variable {ι : Type*} [Fintype ι] {I J : Box ι} {c c₁ c₂ : ℝ≥0} {r r₁ r₂ : (ι → ℝ) → Ioi (0 : ℝ)}
-  {π π₁ π₂ : TaggedPrepartition I}
+variable {ι : Type*} [Fintype ι] {I J : Box ι} {c c₁ c₂ : ℝ≥0}
```
Comment (sgouezel) [inline Mathlib/Analysis/BoxIntegral/Partition/Filter.lean:176]: I don't understand why this change is needed: these are variables that can only be imported if they're mentioned in the type, right?

File: Mathlib/RingTheory/SimpleModule.lean [hunk 2]
```diff
@@ -217,19 +219,21 @@ protected theorem range (f : M →ₗ[R] N) : IsSemisimpleModule R (range f) :=
 
 section
 
-variable [Module S N] {σ : R →+* S} [RingHomSurjective σ] (l : M →ₛₗ[σ] N)
+variable {M'} [AddCommGroup M'] [Module R M'] {N'} [AddCommGroup N'] [Module S N']
```
Comment (sgouezel) [inline Mathlib/RingTheory/SimpleModule.lean:222]:
```suggestion
variable {M' : Type*} [AddCommGroup M'] [Module R M'] {N'} [AddCommGroup N'] [Module S N']
```
Otherwise, I fear it is interpreted as `Type _`, which can (seldom, but still) cause some weird things.

**Review by sgouezel — COMMENTED** [review]
bors d+

File: Mathlib/Analysis/BoxIntegral/Partition/Filter.lean (hunk 1 above)
Comment (semorrison) [reply Mathlib/Analysis/BoxIntegral/Partition/Filter.lean:176]: Yes, it's not strictly necessary. I found it less confusing having these closer to the point of us.
